enhancements/node-latency-profile/node-latency-profiles.md

```diff
@@ -0,0 +1,354 @@
+---
+title: worker-latency-profiles
+authors:
+  - "@harche"
+reviewers:
+  - "@rphillips"
+  - "@sttts"
+  - "@soltysh"
+approvers:
+  - "@rphillips"
+  - "@sttts"
+  - "@soltysh"
+creation-date: 2021-09-28
+last-updated: 2021-10-19
+status: implementable
+see-also:
+  - https://github.com/kubernetes-sigs/kubespray/blob/master/docs/kubernetes-reliability.md
+  - https://github.com/Azure/aks-engine/blob/master/docs/topics/clusterdefinitions.md#controllermanagerconfig
+replaces:
+superseded-by:
+---
+
+# WorkerLatencyProfile
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+
+## Summary
+
+To make sure Openshift cluster keeps running optimally under various circumstances where network latency, API Server latency, etcd latency, control plane load latency, etc. may not always be at their best (e.g. Edge cases), we can tweak the frequency of the status updates done by the `Kubelet` and the corresponding reaction times of the `Controller Manager` and `Kube API Server`.
+
+## Motivation
+
+In order to adjust to different latencies the cluster may experience, we need to adjust how frequently the `Kubelet` updates it's status and how do components such as `Controller Manager` and `Kube API Server`react to those status updates.
+
+In it's more rudimentary form, one can simply edit the relavent arguments to the `Kubelet`, `Controller Manager` and `Kube API Server`to achieve the desired results.
+Upstream `Kubespray` project has excellent [documentation](https://github.com/kubernetes-sigs/kubespray/blob/master/docs/kubernetes-reliability.md#recommendations-for-different-cases) on relavent arguments to each of those components in different scenarios.
+
+But the main motivation of this enhancement is to allow this updates to these critical components in a more controlled manner. Instead of letting the users directly modify those arguments, we want to make sure it happens systematically and methodically. This will eliminate any room for manual errors and would not lead to an unscheduled downtime.
```

Comment:
I thought about it more and before diving more into how to solve the coordination problem I'm missing the entire point why we need one in the first place? What kind of coordination is needed, does it have to be kubelet->kcm->kas or it can be kubelet->kcm+kas? What happens if there's no coordination? The answers to these questions will allow you to choose the most appropriate and effective solution to the problem. I'd like you to come to Tuesday's arch call with these questions answered and we can discuss how to best address them only then.